rbac/management/role/v2_model.py

```diff
@@ -0,0 +1,166 @@
+#
+# Copyright 2025 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Model for role V2 management."""
+
+import uuid_utils.compat as uuid
+from django.db import models
+from django.db.models import signals
+from django.utils import timezone
+from management.models import Permission, Role
+from management.rbac_fields import AutoDateTimeField
+from rest_framework import serializers
+
+from api.models import TenantAwareModel
+
+
+class RoleV2(TenantAwareModel):
+    """V2 Role model."""
+
+    class Types(models.TextChoices):
+        CUSTOM = "custom"
+        SEEDED = "seeded"
+        PLATFORM = "platform"
+
+    uuid = models.UUIDField(default=uuid.uuid7, editable=False, unique=True, null=False)
+    name = models.CharField(max_length=150, null=False, blank=False)
+    description = models.TextField(null=True, blank=True)
+    type = models.CharField(
+        choices=Types.choices, max_length=20, null=False, blank=False, db_index=True, default=Types.CUSTOM
```

Comment:
it looks that this is not validation for model level if type can be any value from  Types. I was able to do:

```
a = RoleV2.objects.create(name="custom_pare4444", tenant=t1, type="anytype")
```

if we will need to cover it, I guess best would to do at db level ? Using check constraint:
https://docs.djangoproject.com/en/5.2/ref/models/constraints/#checkconstraint

```
 class Meta:
        constraints = [
            models.CheckConstraint(
                check=Q(type__in=Types.values),
                name="role_type_check"
            )
        ]
```